File: src/test/java/pageobjects/BurgerRegisterPageTest.java
package pageobjects;

import clients.UserClient;
import io.restassured.response.ValidatableResponse;
import jdk.jfr.Description;
import models.Credentials;
import models.User;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import providers.UserProvider;

import java.util.List;

import static org.apache.http.HttpStatus.SC_OK;

@RunWith(Parameterized.class)
public class BurgerRegisterPageTest {
    private final String name;
    private final String email;
    private final String pwd;
    private final String expected;
    private static final List<String> expecteds = List.of("Некорректный пароль", "Вход");
    private final WebDriver driver;
    private BurgerRegisterPage objBurgerRegisterPage;
    private static UserClient userClient;
    private static User user;
    private static String accessToken;

    @Before
    public void setUp() {
        driver.get("https://stellarburgers.nomoreparties.site/register");
        objBurgerRegisterPage = new BurgerRegisterPage(driver);
    }

    public BurgerRegisterPageTest(String name, String email, String pwd, String expected) {
        driver = new ChromeDriver();
        this.name = name;
        this.email = email;
        this.pwd = pwd;
        this.expected = expected;
    }

    @Parameterized.Parameters
    public static Object[][] getTestData() {
        userClient = new UserClient();
        user = UserProvider.getRandom();
        return new Object[][]{
                {"name", "email", "12345", expecteds.get(0)},
                {user.getName(),user.getEmail(),user.getPassword(), expecteds.get(1)},
        };
    }
    @Test
    @Description("Checking registration with correct/incorrect password")
    public void testRegister() {
        objBurgerRegisterPage.register(name, email, pwd);
        ValidatableResponse responseLogin = userClient.login(Credentials.from(user));
        int statusCode = responseLogin.extract().statusCode();
        if(statusCode == SC_OK){
            accessToken = responseLogin.extract().path("accessToken").toString().substring(6).trim();
        }
        Assert.assertEquals(objBurgerRegisterPage.waitForResultVisibility(expected), expected);
    }
    @After
    public void tearDown() throws InterruptedException {
        driver.quit();
        userClient.delete(accessToken);
        Thread.sleep(300);
    }
}

File: src/test/java/pageobjects/BurgerRegisterPageTest.kt
package pageobjects

import clients.UserClient
import io.restassured.response.ValidatableResponse
import jdk.jfr.Description
import models.Constants.Companion.REGISTER
import models.Credentials.Companion.from
import models.User
import org.apache.http.HttpStatus.SC_OK
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver
import providers.UserProvider



@RunWith(Parameterized::class)
class BurgerRegisterPageTest(private val name: String, private val email: String, private val pwd: String, private val expected: String) {
    init {
            driver = ChromeDriver()
            driver.get(REGISTER)
    }
    @Before
    fun setUp() {
        page = BurgerRegisterPage(driver)
    }
    companion object{
        private lateinit var driver: WebDriver
        private lateinit var page: BurgerRegisterPage
        private lateinit var userClient: UserClient
        private lateinit var user: User
        private lateinit var accessToken: String
        private val expecteds = listOf("Некорректный пароль", "Вход")
        @JvmStatic
        @Parameterized.Parameters
        fun testData(): Collection<Array<Any>> {
            userClient = UserClient()
            user = UserProvider.getRandom()
            return listOf(arrayOf("name", "email", "12345", expecteds[0]), arrayOf(user.name, user.email, user.password, expecteds[1]))
        }
    }

    @Test
    @Description("Checking registration with correct/incorrect password")
    fun testRegister() {
        page.register(name, email, pwd)
        val responseLogin: ValidatableResponse = userClient.login(from(user))
        val statusCode: Int = responseLogin.extract().statusCode()
        if (statusCode == SC_OK) {
            accessToken = userClient.extractToken(from(user))
        }
        Assert.assertEquals(page.waitForResultVisibility(expected), expected)
        if (statusCode == SC_OK) {
            userClient.delete(accessToken)
        }
    }
    @After
    fun tearDown() {
        driver.quit()
        Thread.sleep(300)
    }
}